### ISSUE TITLE
Better syntax coloring

### ISSUE THREAD

**Issue by algorithmax** [notable/notable#1025]
We need better syntax coloring for many programming languages. Maybe this will come in handy.
https://codemirror.net/mode/index.html

**Comment by fabiospampinato** [comment]
This is a bit of a vague request, what should we specifically improve?

**Comment by algorithmax** [comment]
For example, support for gas, nasm language, currently the syntax is not colored.

**Comment by spoorcrystallogy** [comment]
Support for PowerShell would be fantastic, I'm a bit confused reading #811 why it doesn't work at the moment since Monaco does support it?

**Comment by fabiospampinato** [comment]
@spoorcrystallogy by Monaco you mean VS Code?

**Comment by fabiospampinato** [comment]
@spoorcrystallogy I just checked and there's the powershell language defined in the languages list we support, is it not working for you?

**Comment by spoorcrystallogy** [comment]
Sorry, I tried earlier just something elementary (`New-Item -Name bla`) which didn't highlight anything. I pasted in a little script which does show some highlighting, it's just a bit disappointing :( Compare below (github v monarch). 

```powershell
Function Test-IsAdmin {
    <#
    .Synopsis
        Tests if the user is an administrator
    .Description
        Returns true if a user is an administrator, false if the user is not an administrator       
    #>
    $identity = [Security.Principal.WindowsIdentity]::GetCurrent()
    $principal = New-Object Security.Principal.WindowsPrincipal $identity
    $principal.IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator)
}

if (Test-IsAdmin) { 
    Write-Host -ForegroundColor Red "Careful: PowerShell session is elevated." }
else { Write-Host -ForegroundColor Green "PowerShell session is not elevated" }
```
![image](https://user-images.githubusercontent.com/62602956/79051929-47eb4a00-7c33-11ea-9449-31f0548355c0.png)

Unfortunately the [Monarch tester](https://microsoft.github.io/monaco-editor/monarch.html) doesn't have Powershell as an option, but the problem then is that Monarch just isn't very good with Powershell? Not Notables fault, and it's not like I wanted to use it for actual code editing anyway.


> @spoorcrystallogy by Monaco you mean VS Code?
I did mean the Monarch thing but my knowledge about this is 100% based on what was written in #811 😊

**Comment by fabiospampinato** [comment]
> but the problem then is that Monarch just isn't very good with Powershell? 

Yeah, but I'd say it's probably more about the built-in Monarch language definition for PowerShell isn't as good as the one written for the TextMate grammar, I think the grammars that can be written with Monarch are less expressive than what TM allows but I don't know if that's the reason why the built-in grammar Monaco ships with is a lower quality one.

I think at some point we'll have to switch to TM languages too, but that's kind of a big change, and at this point in time it isn't worth spending time making the switch.

**Comment by fabiospampinato** [comment]
It could also be that the language is highlighted well and the syntax highlighting themes I made make a mess of it, but I don't think that's what's happening here.

**Comment by dagadbm** [comment]
could you also add o the cheatsheet which languages are exactly supported or a link to the highlight engine you use so we can see this information?

**Comment by fabiospampinato** [comment]
@dagadbm I'm highlighting things with [Monaco](https://github.com/Microsoft/monaco-editor). In v1.9 there'll be a searchable palette that lists all languages supported:

![image](https://user-images.githubusercontent.com/1812093/82896284-9fc4e280-9f4d-11ea-8f65-a491b80e2b09.png)

Here there's a list of all language IDs:

```
markdown
diff
json
php
abap
aes
apex
azcli
bat
c
clojure
coffeescript
cpp
csharp
csp
css
dockerfile
fsharp
go
graphql
handlebars
html
ini
java
javascript
kotlin
less
lua
mips
msdax
mysql
objective-c
pascal
pascaligo
perl
pgsql
postiats
powerquery
powershell
pug
python
r
razor
redis
redshift
ruby
rust
sb
scheme
scss
shell
sol
sql
st
swift
tcl
twig
typescript
vb
xml
yaml
```

**Comment by dagadbm** [comment]
Thanks :)

Who knows this might just become a good alternative to org mode in
the future
